docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,362 @@
+---
+title: Why Gatsby Uses GraphQL
+---
+
+A common question about Gatsby is, “Why does Gatsby use GraphQL? Doesn’t it generate static files?”
+
+Without providing some context, it can seem like GraphQL is overkill for something like Gatsby. In this document, you’ll see what problems arise when creating pages, and how those problems can be solved using GraphQL.
+
+## Create a page without any data
+
+For any kind of pages that aren’t directly created in `src/pages/`, you’ll need Gatsby’s [`createPages` Node API](/docs/node-apis/#createPages) to create pages programmatically.
+
+All that’s required to create a page is a `path` where it should be created and the component that should be rendered there.
+
+For example, if you had the following component:
+
+```jsx:title=src/templates/no-data.js
+import React from "react"
+
+const NoData = () => (
+  <section>
+    <h1>This Page Was Created Programmatically</h1>
+    <p>
+      No data was required to create this page — it’s just a React component!
+    </p>
+  </section>
+)
+
+export default NoData
+```
+
+You could programmatically create a page at `/no-data/` by adding the following to `gatsby-node.js`:
+
+```js:title=gatsby-node.js
+exports.createPages = ({ actions: { createPage } }) => {
+  createPage({
+    path: "/no-data/",
+    component: require.resolve("./src/templates/no-data.js"),
+  })
+}
+```
+
+After running `npm run develop`, you’ll see the following at `localhost:8000/no-data/`:
+
+![Screenshot of the page generated by the previous code snippet.](./images/why-gql-no-data.png)
+
+In the simplest cases, this is all that’s required for building pages with Gatsby. However, you’ll often want to pass data to the page so that the template component is reusable.
+
+## Create a page with hard-coded data
+
+To pass data to the created pages, you’ll need to pass `context` to the `createPage` call.
+
+In `gatsby-node.js`, we can add context like so:
+
+```js:title=gatsby-node.js
+exports.createPages = ({ actions: { createPage } }) => {
+  createPage({
+    path: "/page-with-context/",
+    component: require.resolve("./src/templates/with-context.js"),
+    context: {
+      title: "We Don’t Need No Stinkin’ GraphQL!",
+      content: "<p>This is page content.</p><p>No GraphQL required!</p>",
+    },
+  })
+}
+```
+
+The `context` property accepts an object, and we can pass in any data we want the page to be able to access.
+
+> **NOTE:** There are a few reserved names that _cannot_ be used in `context`. They are: `path`, `matchPath`, `component`, `componentChunkName`, `pluginCreator___NODE`, and `pluginCreatorId`.
```

Comment:
do we document this anywhere else?

Reply:
Not that I'm aware of — the docs are pretty light on information about `context` in general. @pieh pointed me to the source code, which is where I grabbed these. https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/actions.js#L134-L141

Reply:
It could we good to document those elsewhere, it would work well on this page: https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#pass-context-to-pages

I'll create an issue and PR for it.